src/Nova-GPU-Driver.md

```diff
@@ -1,62 +1,92 @@
+`drivers/gpu/nova-core/` and `drivers/gpu/drm/nova/`
+
+The out-of-tree repository where active development takes place is located on
+[Freedesktop](https://gitlab.freedesktop.org/drm/nova).
+
+
+## Background
+
+### Why a new driver?
+
+Nouveau was, for the most part, designed for pre-GSP hardware. The driver exists
+since ~2009 and its authors back in the day had to reverse engineer a lot about
+the hardware's internals, resulting in a relatively difficult to maintain
+codebase.
+
+Moreover, Nouveau's maintainers concluded that a new driver, exclusively for
+(new) GSP hardware, would allow for significantly simplifying the driver
+design.
 
-In the source tree, the driver lives in `drivers/gpu/drm/nova`.
 
+### Why write it in Rust?
 
-## Status
+Rust's most desired feature are its guarantees for memory safety, notably the
+elimination of Use-after-Free errors. Those are errors GPU drivers suffer from
+significantly, because GPUs are, by definition, asynchronous in regards to the
+CPU and can handle millions of jobs (i.e., memory objects) simultaneously. Jobs'
+status can be changed at different places in the code base at different
+points in time (through work items, interrupt handlers, userspace calls, ...).
 
-Currently, Nova is just a stub driver intended to lift the bindings necessary
-for a real GPU driver into the (mainline) kernel.
+In short, GPU drivers were expected to profit the most from the promised memory
+safety.
 
-Currently, those efforts are mostly focused on getting bindings for PCI, DRM
-and the Device (driver) model upstream.
+Since Nova was a freshly written new driver, it was an opportunity to try to
+leverage the advantages of Rust and obtain a more reliable, maintainable driver.
 
-It can be expected that, as the driver continues to grow, various other abstractions
-will be needed.
+Another reason is that Rust's macros are helpful when creating firmware bindings
+for continuously changing firmware versions and their ABIs.
 
 
-## Utilized Common Rust Infrastructure
+## Architecture
 
-Nova depends on the Rust for Linux `staging/*` [branches](Branches.md).
+![Nova Architecture with vGPUs](./nova-core-vm.png)
 
+The overall GPU driver is split into two parts:
 
-## Contributing
+1. "Nova-Core", living in `linux/drivers/gpu/nova-core/`. Nova-Core implements
+   the fundamental interaction with the hardware (through PCI etc.) and,
+   notably, the firmware interfaces necessary to communicate with an Nvidia
+   GPU.
+2. "Nova-DRM" (the official name is actually just "Nova", but to avoid
+   confusion developers usually call it "Nova-DRM"), living in
+   `linux/drivers/gpu/drm/nova/`. This is the actual graphics driver,
+   implementing all the typical DRM interfaces for userspace.
 
-As with every real open source program, help and participation is highly welcome!
+This split architecture allows for virtualizing GPUs: Nova-Core can be used to
+instruct the GPU's firmware to spawn a new PCI subdevice, which can then be
```

Comment:
You may want to drop just a few words about SR-IOV and virtual functions [1].

[1] https://docs.kernel.org/PCI/pci-iov-howto.html